packages/next/src/server/after/after-context.ts

```diff
@@ -0,0 +1,200 @@
+import { DetachedPromise } from '../../lib/detached-promise'
+import {
+  requestAsyncStorage,
+  type RequestStore,
+} from '../../client/components/request-async-storage.external'
+import { BaseServerSpan } from '../lib/trace/constants'
+import { getTracer } from '../lib/trace/tracer'
+import type { CacheScope } from './react-cache-scope'
+import { ResponseCookies } from '../web/spec-extension/cookies'
+import type { RequestLifecycleOpts } from '../base-server'
+import type { AfterCallback, AfterTask, WaitUntilFn } from './shared'
+
+export type AfterContext =
+  | ReturnType<typeof createAfterContext>
+  | ReturnType<typeof createDisabledAfterContext>
```

Comment:
nit: Would be nicer to have actual named, explicitly typed types to remove this indirection.

Reply:
Added an explicit interface + removed the "DisabledAfterContext" variant, so this is cleaner now